lib/activerecord-tenant-level-security/schema_dumper.rb

```diff
@@ -28,23 +29,41 @@ def policies_in_database
           tablename;
       SQL
       results = ActiveRecord::Base.connection.execute(query)
-      table_names = results.map { |x| x["tablename"] }
+      results.map do |result|
+        table_name = result["tablename"]
+        partition_key = convert_qual_to_partition_key(result["qual"])
+        Policy.new(table_name: table_name, partition_key: partition_key)
+      end
+    end
+
+    private
+
+    def convert_qual_to_partition_key(qual)
+      matched = qual.match(/^\((.+?) = /)
+      # This error can occur if the specification of the 'tenant_policy' in PostgreSQL
+      #   or the 'create_policy' method changes
+      raise "Failed to parse partition key from 'qual': #{qual}" unless matched
 
-      table_names.map { |t| Policy.new(t) }
+      matched[1]
     end
 
     class Policy
-      def initialize(table_name)
+      def initialize(table_name:, partition_key:)
         @table_name = table_name
+        @partition_key = partition_key
       end
 
       def to_schema
-        %(  create_policy "#{table_name}")
+        schema = %(  create_policy "#{table_name}")
+        if partition_key && partition_key != TenantLevelSecurity::DEFAULT_PARTITION_KEY
```

Comment:
If the `partition_key` is always present, the condition can be simplified to:

```suggestion
        if partition_key != TenantLevelSecurity::DEFAULT_PARTITION_KEY
```

Reply:
You are right, I added it https://github.com/kufu/activerecord-tenant-level-security/pull/22/commits/35025e4028438951f9f108f57270808dfae713f5